src/intervals/arithmetic/integer.jl

```diff
@@ -2,56 +2,112 @@
 # 9.1 of the IEEE Standard 1788-2015 and required for set-based flavor in
 # Section 10.5.3
 
-for f ∈ (:sign, :ceil, :floor, :trunc)
-    @eval begin
-        """
-            $($f)(a::BareInterval)
-            $($f)(a::Interval)
-
-        Implement the `$($f)` function of the IEEE Standard 1788-2015 (Table 9.1).
-        """
-        function $f(a::BareInterval{T}) where {T<:NumTypes}
-            isempty_interval(a) && return a
-            lo, hi = bounds(a)
-            return _unsafe_bareinterval(T, $f(lo), $f(hi))
-        end
+"""
+    sign(x::BareInterval)
+    sign(x::Interval)
 
-        function $f(a::Interval)
-            r = $f(bareinterval(a))
-            d = decoration(a)
-            d = min(d, ifelse(isthin(r), d, def))
-            return _unsafe_interval(r, d, isguaranteed(a))
-        end
-    end
+Implement the `sign` function of the IEEE Standard 1788-2015 (Table 9.1).
+"""
```

Comment:
Is this change motivated by the docstrings? If so, the docs can be included afterwords using `@docs """...""" fn_name`. Seems to me that the code exploiting metaprogramming is more concise.